test/val/val_id_test.cpp

```diff
@@ -1848,6 +1848,30 @@ OpFunctionEnd
   EXPECT_EQ(SPV_SUCCESS, ValidateInstructions());
 }
 
+TEST_F(ValidateIdWithMessage, OpVariableContainsOpTypeBoolBad) {
```

Comment:
Please add some positive tests. Also make sure to cover the builtin exclusion you added.